model/config.go

```diff
@@ -25,19 +27,32 @@ import (
 	"istio.io/pilot/model/test"
 )
 
-// Config is a configuration unit consisting of the type of configuration, the
-// key identifier that is unique per type, and the content represented as a
-// protobuf message.  The revision is optional, and if provided, identifies the
+// ConfigMeta is metadata attached to each configuration unit.
+// The revision is optional, and if provided, identifies the
 // last update operation on the object.
-type Config struct {
+type ConfigMeta struct {
 	// Type is a short configuration name that matches the content message type
-	Type string
+	// (e.g. "route-rule")
```

Comment:
Are type names normalized to non-k8s specific names? e.g. route-rule instead of route-rule.config.istio.io?